File: September 2024/September 27/chocolate_feast.py
import math

def chocolate_feast(n, c, m):
    bars = 0
    wrappers = 0

    bars = math.floor(n/c)
    wrappers = bars

    while wrappers >= m:
        add = math.floor(wrappers/m)
        bars += add
        wrappers -= (add * m)
        wrappers += add

    
    return bars

print(chocolate_feast(15,3,2))

File: September 2024/September 27/chocolateFeast.js
function chocolateFeast(n, c, m) {
    let bars = 0;
    let wrappers = 0;
  
    bars = Math.floor(n / c);
    wrappers = bars;
  
    while (wrappers >= m) {
      let add = Math.floor(wrappers / m)
      bars += add;
      wrappers -= add * m;
      wrappers += add;
    }
  
    return bars;
  }